temba/api/v2/views.py

```diff
@@ -3573,3 +3581,454 @@ def post(self, request, *args, **kwargs):
+            return Response(
+                {
+                    "error": _(
+                        "Please make sure the collection name only contains "
+                        "alphanumeric characters [0-9a-zA-Z], spaces, underscores and hyphens"
+                    )
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        collection = self.get_collection_full_name(org=org, collection=collection_name)
+        url = f"{settings.PARSE_URL}/schemas/{collection}"
+        data = {
+            "className": collection,
+            "fields": DEFAULT_FIELDS_PAYLOAD_LOOKUPS,
+            "indexes": DEFAULT_INDEXES_FIELDS_PAYLOAD_LOOKUPS,
+        }
+        response = requests.post(url, data=json.dumps(data), headers=self.parse_headers)
+        if response.status_code == 200:
+            org.add_collection_to_org(self.request.user, collection_name, collection_type=LOOKUPS)
+        else:
+            return Response(response.json(), status=response.status_code)
+
+        return Response(status=status.HTTP_201_CREATED)
+
+    def delete(self, request, *args, **kwargs):
+        org, collection_name, collections_list, error_response = self.get_default_params(config="collection_exist")
+        if error_response:
+            return error_response
+
+        collection = self.get_collection_full_name(org=org, collection=collection_name)
+        purge_url = f"{settings.PARSE_URL}/purge/{collection}"
+        url = f"{settings.PARSE_URL}/schemas/{collection}"
+
+        try:
+            collection_index = collections_list.index(collection_name)
+        except ValueError:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        response_purge = requests.delete(purge_url, headers=self.parse_headers)
+        if response_purge.status_code in [200, 404]:
+            response = requests.delete(url, headers=self.parse_headers)
+
+            if response.status_code == 200:
+                org.remove_collection_from_org(user=self.request.user, index=collection_index, collection_type=LOOKUPS)
+                return Response(status=status.HTTP_204_NO_CONTENT)
+            else:
+                return Response(response.json(), status=response.status_code)
+        else:
+            return Response(response_purge.json(), status=response_purge.status_code)
+
+    def put(self, request, *args, **kwargs):
+        org, collection_name, collections_list, error_response = self.get_default_params(config="collection_exist")
+        if error_response:
+            return error_response
+
+        fields_to_skip = ["objectId", "createdAt", "updatedAt", "ACL"]
+        fields_to_create = dict(
+            filter(lambda x: x[0] not in fields_to_skip, self.request.data.get("fields", {}).items())
+        )
+        items_to_push = self.request.data.get("items", [])
+        if not fields_to_create and not items_to_push:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Let's add a message to this response, this way the user would know what's wrong.

Reply:
Added response details.